client/lib/load-script/index.js

```diff
@@ -3,92 +3,186 @@
  */
 
 /**
- * Internal dependency.
+ * External dependencies
  */
-var config = require( 'config' );
+import debugFactory from 'debug';
+const debug = debugFactory( 'lib/load-script' );
 
-var JQUERY_URL = 'https://s0.wp.com/wp-includes/js/jquery/jquery.js',
-	callbacksForURLsInProgress = {};
+/**
+ * Internal dependencies
+ */
+import config from 'config';
 
-var loadScript = function( url, callback ) {
-	var script = document.createElement( 'script' ),
-		loaded = false;
+/**
+ * Module variables
+ */
+const callbacksForURLsInProgress = {};
```

Comment:
Why not use proper `Map` for it? `delete` is [slow](https://stackoverflow.com/questions/43594092/slow-delete-of-object-properties-in-js-in-v8).

Reply:
@yurynix Why not yeah. It's as readable imo.
`delete` is super fast compared to the round trip time of an http request so probably not worth micro-optimizing a few `delete` calls.